### PR TITLE
Dust selection bug

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
ontop of #1185 to avoid conflicts

- adds a failing test when dust is selected
- comments out dust selection for now

### CODE REVIEW

**Comment by zancas** [comment]
I clicked update on this PR.

File: libtonode-tests/tests/legacy.rs [hunk 1]
```diff
@@ -4244,3 +4245,26 @@ mod basic_transactions {
 async fn proxy_server_worky() {
     zingo_testutils::check_proxy_server_works().await
 }
+
```
Comment (zancas) [inline libtonode-tests/tests/legacy.rs:4248]: I see this test will send dust too sapling from Orchard.   I'm not clear on correct behavior.

**Comment by zancas** [comment]
I vote that we accept the amount leaked (1-dust, i.e. < 5_000) and the amount demotion (again < 5_000) in order to rescue that same amount, and have a less complex policy.

**Comment by zancas** [comment]
In other words, I think it's a useful test highlighting an interesting case.   Probably the best balance of concerns is to take the small demotion and information leak, for the payoff of a relatively simple sweeping policy, and rescue (albeit demoted) of the dust.

**Comment by Oscar-Pepper** [comment]
closed for #1188

**Pull request closed without merging**